usr/share/rear/build/GNU/Linux/100_copy_as_is.sh

```diff
@@ -59,7 +59,7 @@ Log "Adding required libraries of executables in all the copied files to LIBS"
 local required_library=""
 for required_library in $( RequiredSharedOjects "${copy_as_is_executables[@]}" ) ; do
     # Skip when the required library was already actually copied by 'tar' above:
-    grep -q "$required_library" $copy_as_is_filelist_file && continue
+    egrep -q "^$required_library$" $copy_as_is_filelist_file && continue
```

Comment:
@rmetrich why not using `grep -E` instead of `egrep` - less overhead,

Reply:
Well, because I never use "grep -E" ;-)
Fixed.

Reply:
@rmetrich 
I wonder why extended regular expressions `-E` are neded at all here?
For me it works with plain grep:
<pre>
# export pattern="bar"

# ( set -x ; echo "foobar" | grep "^$pattern$" )
+ echo foobar
+ grep '^bar$'

# ( set -x ; echo "barbaz" | grep "^$pattern$" )
+ echo barbaz
+ grep '^bar$'

# ( set -x ; echo "bar" | grep "^$pattern$" )
+ echo bar
+ grep '^bar$'
bar
</pre>

I wonder if the unescaped plain `$` at the end is fully safe in any bash.
For me it works but shouldn't it be better `\$` to be more on the safe side?

Reply:
Well on Linux it indeed works (i.e. with GNU grep). There is a lot of `grep` vs `egrep` commands in ReaR. Usually when there is a regex, `egrep` is used, but not always.
Let's go for basic "grep" and "\\$" then.